From 77fa552b4c7c636e22437aa8ce2a74f5278e0004 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 060/296] Crash occured due to null pointer exception.

Reason: Trying to access remote device
object (mobjectdevice)which have been set to NULL before.

Fix:Avoid handling state changes for invalid devices(NULL)
---
 .../android/settingslib/bluetooth/LocalBluetoothProfileManager.java  | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/packages/SettingsLib/src/com/android/settingslib/bluetooth/LocalBluetoothProfileManager.java b/packages/SettingsLib/src/com/android/settingslib/bluetooth/LocalBluetoothProfileManager.java
index 0750dc75005..88cbf435366 100755
--- a/packages/SettingsLib/src/com/android/settingslib/bluetooth/LocalBluetoothProfileManager.java
+++ b/packages/SettingsLib/src/com/android/settingslib/bluetooth/LocalBluetoothProfileManager.java
@@ -296,6 +296,11 @@ public class LocalBluetoothProfileManager {
         }
 
         public void onReceive(Context context, Intent intent, BluetoothDevice device) {
+            if (device == null) {
+                if(DEBUG) Log.d(TAG, "StateChangedHandler receives state-change for invalid device");
+                return;
+            }
+
             CachedBluetoothDevice cachedDevice = mDeviceManager.findDevice(device);
             if (cachedDevice == null) {
                 Log.w(TAG, "StateChangedHandler found new device: " + device);
-- 
2.11.0

